File: DeJia/app/src/main/java/com/dejia/anju/view/MyPullRefresh.java
package com.dejia.anju.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import com.qmuiteam.qmui.widget.pullRefreshLayout.QMUIFollowRefreshOffsetCalculator;
import com.qmuiteam.qmui.widget.pullRefreshLayout.QMUIPullRefreshLayout;

public class MyPullRefresh extends QMUIPullRefreshLayout {

    private RefreshListener refreshListener;

    public MyPullRefresh(Context context) {
        this(context, null);
    }

    public MyPullRefresh(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MyPullRefresh(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        //设置下拉后位置
        setRefreshOffsetCalculator(new QMUIFollowRefreshOffsetCalculator());
        setOnPullListener(new OnPullListener() {
            @Override
            public void onMoveTarget(int offset) {

            }

            @Override
            public void onMoveRefreshView(int offset) {

            }

            @Override
            public void onRefresh() {
                if (refreshListener != null) {
                    refreshListener.onRefresh();
                }
            }
        });
    }

    /**
     * 设置自定义的刷新头
     *
     * @return
     */
    @Override
    protected View createRefreshView() {
        return new CustomRefresh(getContext());
    }


    /**
     * @param listener
     */
    public void setRefreshListener(RefreshListener listener) {
        this.refreshListener = listener;
    }


    public interface RefreshListener {
        void onRefresh();
    }
}

File: DeJia/app/src/main/java/com/dejia/anju/view/MyPullRefresh.kt
package com.dejia.anju.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import com.qmuiteam.qmui.widget.pullRefreshLayout.QMUIFollowRefreshOffsetCalculator
import com.qmuiteam.qmui.widget.pullRefreshLayout.QMUIPullRefreshLayout

class MyPullRefresh @JvmOverloads constructor(context: Context?, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : QMUIPullRefreshLayout(context, attrs, defStyleAttr) {
    private var refreshListener: RefreshListener? = null

    /**
     * 设置自定义的刷新头
     *
     * @return
     */
    override fun createRefreshView(): View {
        return CustomRefresh(context)
    }

    /**
     * @param listener
     */
    fun setRefreshListener(listener: RefreshListener?) {
        refreshListener = listener
    }

    interface RefreshListener {
        fun onRefresh()
    }

    init {
        //设置下拉后位置
        setRefreshOffsetCalculator(QMUIFollowRefreshOffsetCalculator())
        setOnPullListener(object : OnPullListener {
            override fun onMoveTarget(offset: Int) {}
            override fun onMoveRefreshView(offset: Int) {}
            override fun onRefresh() {
                if (refreshListener != null) {
                    refreshListener!!.onRefresh()
                }
            }
        })
    }
}